app/controllers/concerns/two_factor_authenticatable.rb

```diff
@@ -46,12 +46,12 @@ def handle_valid_otp
     redirect_to after_sign_in_path_for(current_user)
   end
 
-  def handle_invalid_otp
+  def handle_invalid_otp(type: 'otp')
```

Comment:
since the method name has `_otp` in it, maybe rename method, or add a comment about what other valid `type` values are supported?